Doc/library/typing.rst

```diff
@@ -1565,6 +1741,75 @@ These are not used in annotations. They are building blocks for creating generic
    .. versionadded:: 3.10
 
 
+.. class:: TypeAliasType(name, value, *, type_params=())
+
+   The type of type aliases created through the :keyword:`type` statement.
+
+   Example::
+
+      >>> type Alias = int
+      >>> type(Alias)
+      <class 'typing.TypeAliasType'>
+
+   .. versionadded:: 3.12
+
+   .. attribute:: __name__
+
+      The name of the type alias.
+
+      Example::
+
+         >>> type Alias = int
+         >>> Alias.__name__
+         'Alias'
+
+   .. attribute:: __module__
+
+      The module in which the type alias was defined.
+
+      Example::
+
+         >>> type Alias = int
+         >>> Alias.__module__
+         '__main__'
+
+   .. attribute:: __type_params__
+
+      The type parameters of the type alias, or an empty tuple if the alias is
+      not generic.
+
+      Example::
+
+         .. doctest::
+
+         >>> type ListOrSet[T] = list[T] | set[T]
+         >>> ListOrSet.__type_params__
+         (T,)
+         >>> type NotGeneric = int
+         >>> NotGeneric.__type_params__
+         ()
```

Comment:
This isn't the correct syntax for the `.. doctest` directive; it isn't rendering correctly: https://cpython-previews--104642.org.readthedocs.build/en/104642/library/typing.html#typing.TypeAliasType.__type_params__

```suggestion
      The type parameters of the type alias, or an empty tuple if the alias is
      not generic:

      .. doctest::

          >>> type ListOrSet[T] = list[T] | set[T]
          >>> ListOrSet.__type_params__
          (T,)
          >>> type NotGeneric = int
          >>> NotGeneric.__type_params__
          ()
```